disputer/disputer.js

```diff
@@ -15,6 +23,41 @@ class Disputer {
 
     // Instance of the expiring multiparty to perform on-chain disputes
     this.empContract = this.empClient.emp;
+
+    // Default config settings. Disputer deployer can override these settings by passing in new
+    // values via the `config` input object. The `isValid` property is a function that should be called
+    // before resetting any config settings. `isValid` must return a Boolean.
+    const defaultConfig = {
+      disputeDelay: {
+        // `disputeDelay`: Amount of time to wait after the request timestamp of the liquidation to be disputed.
+        // This makes the reading of the historical price more reliable. Denominated in seconds.
+        value: 60,
+        isValid: x => {
+          return x >= 0;
+        }
+      },
+      txnGasLimit: {
+        // `txnGasLimit`: Gas limit to set for sending on-chain transactions.
+        value: 9000000, // Can see recent averages here: https://etherscan.io/chart/gaslimit
+        isValid: x => {
+          return x >= 6000000 && x < 15000000;
+        }
+      }
+    };
+
+    // Set and validate config settings
+    Object.keys(defaultConfig).forEach(field => {
+      this[field] = config && Object.keys(config).includes(field) ? config[field] : defaultConfig[field].value;
+      if (!defaultConfig[field].isValid(this[field])) {
+        this.logger.error({
+          at: "Disputer",
+          message: "Attempting to set configuration field with invalid value",
+          field: field,
+          value: this[field]
+        });
+        throw new Error("Attempting to set configuration field with invalid value");
+      }
+    });
```

Comment:
This logic is super useful -- maybe add a TODO to abstract this general logic out to a common function so we don't have to include it everywhere we do this sort of config processing (my guess is that we'll end up using this in more places).

Reply:
I'll add the TODO comment in this PR and abstract it out in a follow up